ethers-core/src/types/transaction/optimism.rs

```diff
@@ -48,14 +48,14 @@ impl DepositTransaction {
         let mut rlp = RlpStream::new();
         rlp.begin_list(NUM_TX_FIELDS);
 
-        rlp_opt(&mut rlp, &self.source_hash);
-        rlp.append(&self.tx.from);
+        rlp.append(&self.source_hash.unwrap_or_default());
+        rlp.append(&self.tx.from.unwrap_or_default());
```

Comment:
ah yes, we need to unwrap the from field because deposit txs don't have a signature...